lib/srv/desktop/rdp/rdpclient/src/rdpdr/path.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::ffi::{CString, NulError};
+
+/// WindowsPath is a String that we assume to be in the form
+/// of a traditional DOS path:
+///
+/// https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats
+///
+/// Because RDP device redirection is limited in the paths it uses, we can
+/// further assume that it is in one of the following forms:
+///
+/// r"\Program Files\Custom Utilities\StringFinder.exe": An absolute path from the root of the current drive.
+///
+/// r"2018\January.xlsx": A relative path to a file in a subdirectory of the current directory.
+#[derive(Debug, Clone)]
+pub struct WindowsPath {
+    path: String,
+}
+
+impl WindowsPath {
+    pub fn new(path: String) -> WindowsPath {
```

Comment:
Maybe this could be a function with no parameters that gives you an empty path (much like `String::new`) and the construction from a string should just be a `From<String>` impl?

Reply:
23cf137788b341d89fb48cb0eeacbc4904d7d46e